server/src/internalClusterTest/java/org/elasticsearch/action/admin/indices/rollover/RolloverIT.java

```diff
@@ -154,7 +154,7 @@ public void testRolloverWithIndexSettings() throws Exception {
         indexDoc("test_index-2", "1", "field", "value");
         flush("test_index-2");
         final Settings settings = Settings.builder()
-            .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+            .put("number_of_shards", 1) // testing without "index" prefix
```

Comment:
I think I'd prefer a dedicated test for this, just so that it doesn't get accidentally removed and the behavior reverted in the future, could you add a test for this? (it can basically be a copy of this test with a descriptive name)